lib/egg.js

```diff
@@ -351,22 +351,54 @@ class EggCore extends KoaApplication {
   }
 
   /**
-   * use co to wrap generator function to a function return promise
-   * @param  {GeneratorFunction} fn input function
-   * @return {AsyncFunction} async function return promise
+   * Convert a generator function to a promisable one.
+   *
+   * Notice: for other kinds of functions, it directly returns you what it is.
+   *
+   * @exception If `fn` isn't a function, error will be thrown out.
+   * @param  {Function} fn The inputted function.
+   * @param {Boolean} strictMode When this is set to `true`, your `fn` must be a function,
```

Comment:
is this a explicit use case for strictMode now? if not, maybe we should not add this option.

Reply:
To be honest, toAsyncFunction's parameter MUST BE function. And we didn't do validations to that before. Now you can reguard this as a pack to be compatible with what it was before. And I'll rewrite the comment of `@exception`, and this will ONLY happen when your strictMode = true.

Reply:
The answer is : YES.
To be honest, toAsyncFunction's parameter MUST BE function. And we didn't do validations to that before. Now you can reguard this as a pack to be compatible with what it was before. And `strictMode` is an optional param.

Reply:
@dead-horse：Do you mean for an optional param that I shouldn't show it in the comment?

Reply:
I'll remove this feature according to altian's idea, please refer: https://github.com/eggjs/egg/issues/2931